### DISCUSSION TITLE
Markdown: How to suppress auto-generated markdown anchors?

### DISCUSSION THREAD

**Discussion by alegrigoriev** [community/community#70734] [Programming Help] [answered]
### Body

When Github renders Markdown (.md) files, it auto-generates HTML anchors for every heading and shows the "chain links" icon next to the heading line.

How do I suppress such auto-generated anchors, and only show the "links" icon for the headings I explicitly add an anchor, as in this markdown source:

```
XML configuration file {#xml-config-file}
======================
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by arienshibani** [comment, marked as answer]
GitHub automatically generates anchors for each heading in a Markdown file, and there's no built-in way to suppress this behavior. The "chain links" icon is also automatically added to each heading when you hover over it, and this behavior can't be suppressed either. However, you can manually add anchors to your headings using HTML syntax, and these will override the automatically generated anchors. Here's how you can do it:

```md
## <a id="custom-id"></a>My Heading
```

In this example, `My Heading` will have the anchor `#custom-id` instead of `#my-heading`. The "chain links" icon will still appear when you hover over the heading, but it will link to `#custom-id`. If you want to hide the "chain links" icon entirely, you would need to use custom CSS, but GitHub doesn't allow custom CSS in Markdown files for security reasons. So unfortunately, there's no way to hide the "chain links" icon for certain headings on GitHub.

**Reply by alegrigoriev** [reply]
Thanks. That's a disappointment. Another example of too helpful software.

Just so you know, you can add an explicit anchor as:
```
## My Heading{#custom-id}
```

**Comment by Abhijitkr** [comment]
Those anchor links are shown only when html tags are used, instead use Markdown language like replace your heading html tags with # , html <a></a>tags with ![Title here](Link here) etc.

**Reply by alegrigoriev** [reply]
Not in my experience. I have a file without any HTML tags, the links are still shown.
